src/viam/ur/module/test.cpp

```diff
@@ -13,8 +13,57 @@ Eigen::VectorXd makeVector(std::vector<double> data) {
     return Eigen::Map<Eigen::VectorXd>(data.data(), boost::numeric_cast<Eigen::Index>(data.size()));
 }
 
+// trajectory_sample_point
 }  // namespace
 
+BOOST_AUTO_TEST_CASE(test_sampling_func) {
+    // test a random set of samples
+    {
+        std::vector<trajectory_sample_point> test_samples = {};
+        const double test_duration_sec = 5;
+        const double test_freq_hz = 5;
+        // we are not testing the behavior of the trajectory generation, so we create a simple lambda function.
+        // this allows us to get a good idea of what sampling_func is doing
+        sampling_func(test_samples, test_duration_sec, test_freq_hz, [](const double t, const double step) {
+            return trajectory_sample_point{{t, 0, 0, 0, 0, 0}, {t * step, 0, 0, 0, 0, 0}, boost::numeric_cast<float>(step)};
+        });
+        BOOST_CHECK_EQUAL(test_samples.size(), 26);
```

Comment:
could probably make this `BOOST_CHECK_EQUAL(test_samples.size(), test_duration_sec*test_freq_hz + 1)` so it doesn't seem like a magic number. will be easy to adjust too pending on if we include t0